File: Python/1281. Subtract the Product and Sum of Digits of an Integer.py
'''
1281. Subtract the Product and Sum of Digits of an Integer
Problem Link : https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/
Runtime: 8 ms
Memory Usage: 13.3 MB
'''
class Solution(object):
    def subtractProductAndSum(self, n):
        m=1
        s=0
        a=list(str(n))
        for i in range(len(a)):
            m*=int(a[i])
            s+=int(a[i])
        return m-s

File: java/1281 Subtract the Product and Sum of Digits of an Integer.java
/*
1281. Subtract the Product and Sum of Digits of an Integer
**********
Problem Link : https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/
Runtime: 0 ms
Memory Usage: 38.7 MB
**********
*/
class Solution {
    public int subtractProductAndSum(int n) {
            int rem=0;
	        int multi=1,sum=0;
              while(n>0)
              {
                  rem=n%10;
                  multi=rem*multi;
                  sum=rem+sum;
                  n=n/10;
              }
        return multi-sum;
    }
}
